### DISCUSSION TITLE
npm run:publish not working on the create-dapp-template Minting NFT

### DISCUSSION THREAD

**Discussion by GSgiansen** [aptos-labs/aptos-developer-discussions#400] [Questions] [answered]
Hello,

I've recently tried to use the create-dapp-template and chose the Minting NFTs template. 
After following the guide provided at https://aptos.dev/en/build/create-aptos-dapp/templates/nft-minting-dapp, I am able to run `npm run move:init` but whenever i run `npm run move:publish`, this error keeps occurring for me.

<img width="1268" alt="Screenshot 2024-09-04 at 3 59 23 PM" src="https://github.com/user-attachments/assets/49318cb1-d490-4cec-823c-646200cea602">

Could I check if this error occurs for anyone else ? and if so how can I go about fixing this ?

**Comment by ajaythxkur** [comment, marked as answer]
Have you updated env??
`VITE_APP_NETWORK=testnet
VITE_COLLECTION_CREATOR_ADDRESS=0xC0FEE`

**Reply by GSgiansen** [reply]
yeap, ive followed that step to edit the collector creator address to my own wallet address

**Reply by ajaythxkur** [reply]
`{
  "Error": "Unexpected error: Unable to resolve packages for package 'digital_asset_launchpad': While resolving dependency 'AptosFramework' in package 'digital_asset_launchpad': Failed to reset to latest Git state 'mainnet' for package 'AptosFramework', to skip set --skip-fetch-latest-git-deps | Exit status: exit status: 128"
}`
Did you see this error in cli?

**Reply by GSgiansen** [reply]
Yes this error occurs when I go into the move folder and compile it from there

**Reply by ajaythxkur** [reply]
@GSgiansen Add an extra argument in your `createObjectAndPublishPackage` function in `scripts/move/publish.js`
`extraArguments: ["--skip-fetch-latest-git-deps"]`

e.g.
`createObjectAndPublishPackage({
      packageDirectoryPath: "move",
      addressName: "launchpad_addr",
      namedAddresses: {
        launchpad_addr: accountAddress,
        initial_creator_addr: process.env.VITE_COLLECTION_CREATOR_ADDRESS,
        minter: "0x3c41ff6b5845e0094e19888cba63773591be9de59cafa9e582386f6af15dd490",
      },
      profile: `${process.env.PROJECT_NAME}-${process.env.VITE_APP_NETWORK}`,
      extraArguments: ["--skip-fetch-latest-git-deps"]
    })`

**Reply by GSgiansen** [reply]
Thank you so much it managed to work!
